e2e/container/kill_test.go

```diff
@@ -0,0 +1,78 @@
+package container
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/cli/e2e/internal/fixtures"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKillContainer(t *testing.T) {
+	containerID := runBackgroundTop(t)
+
+	// Kill with a signal should send the signal to the process
+	// if the signal is not a deadly one, the container should still live on
+	result := icmd.RunCmd(
+		icmd.Command("docker", "kill", "-s", "SIGWINCH", containerID),
+	)
+	result.Assert(t, icmd.Success)
+
+	result = icmd.RunCmd(
+		shell(t, "docker inspect -f '{{ .State.Status }}' %s", containerID),
+	)
+	result.Assert(t, icmd.Success)
+	require.Equal(t, "running", strings.TrimSpace(result.Stdout()))
+
+	// Kill with no signal should kill the process
+	result = icmd.RunCmd(
+		icmd.Command("docker", "kill", containerID),
+	)
+
+	result.Assert(t, icmd.Success)
+	poll.WaitOn(t, containerStatus(t, containerID, "exited"), poll.WithDelay(1*time.Second), poll.WithTimeout(10*time.Second))
+
+	// Kill on a stop container should return an error
+	result = icmd.RunCmd(
+		icmd.Command("docker", "kill", containerID),
+	)
+	result.Assert(t, icmd.Expected{
+		ExitCode: 1,
+	})
+}
+
+func TestKillRunningContainer(t *testing.T) {
```

Comment:
This test is a little redundant with the previous one, no?

Reply:
arf this one was supposed to be gone yep 😛